### DISCUSSION TITLE
differentiation of composite functions

### DISCUSSION THREAD

**Discussion by fabpan** [sympy/sympy#22919] [Q&A]
Hi,

I'm trying to manipulate this expression 

<img src="https://render.githubusercontent.com/render/math?math=\displaystyle g^{2}{\left(y{\left(t \right)},t \right)} \frac{d^{2}}{d y{\left(t \right)}^{2}} g{\left(y{\left(t \right)},t \right)} + g{\left(y{\left(t \right)},t \right)} \left. \frac{\partial^{2}}{\partial t\partial \xi} g{\left(\xi,t \right)} \right|_{\substack{ \xi=y{\left(t \right)} }} + g{\left(y{\left(t \right)},t \right)} \left. \frac{\partial^{2}}{\partial y{\left(t \right)}\partial \xi_{2}} g{\left(y{\left(t \right)},\xi_{2} \right)} \right|_{\substack{ \xi_{2}=t }} + \frac{d}{d t} g{\left(y{\left(t \right)},t \right)} \frac{d}{d y{\left(t \right)}} g{\left(y{\left(t \right)},t \right)} + \left. \frac{\partial^{2}}{\partial \xi_{2}^{2}} g{\left(y{\left(t \right)},\xi_{2} \right)} \right|_{\substack{ \xi_{2}=t }}" />

that I get from the following code on notebook:

```
import sympy as sy
t=sy.symbols('t')
y=sy.Function('y')(t)
g=sy.Function('g')(y,t)
exp=sy.diff(g,t,2)
exp1=exp.subs([(sy.diff(y,t),g)]).expand()
display(exp1)
```

The question is how to get the substitutions indicated by the bars made.
They are not made by ```exp1.doit()```
As result I would expect:
<img src="https://render.githubusercontent.com/render/math?math=\displaystyle g^{2}{\left(y{\left(t \right)},t \right)} \frac{d^{2}}{d y{\left(t \right)}^{2}} g{\left(y{\left(t \right)},t \right)} + 2 g{\left(y{\left(t \right)},t \right)}  \frac{\partial^{2}}{\partial t\partial y(t)} g{\left(y(t),t \right)}  + \frac{d}{d t} g{\left(y{\left(t \right)},t \right)} \frac{d}{d y{\left(t \right)}} g{\left(y{\left(t \right)},t \right)} + \frac{\partial^{2}}{\partial t^{2}} g{\left(y{\left(t \right)},t \right)}" />

**Comment by oscarbenjamin** [comment]
This is a bug, probably in `Derivative._eval_subs`
